pkg/interface/src/views/components/OverlaySigil.tsx

```diff
@@ -1,134 +1,103 @@
-  }
-
-  profileShow() {
-    this.updateContainerOffset();
-    this.setState({ clicked: true });
-    this.props.scrollWindow?.addEventListener('scroll', this.updateContainerOffset);
-  }
-
-  profileHide() {
-    this.setState({ clicked: false });
-    this.props.scrollWindow?.removeEventListener('scroll', this.updateContainerOffset, true);
-  }
-
-  updateContainerOffset() {
-    if (this.containerRef && this.containerRef.current) {
-      const container = this.containerRef.current;
-      const scrollWindow = this.props.scrollWindow;
-      const bottomSpace =  scrollWindow ? scrollWindow.clientHeight - ((container.getBoundingClientRect().top + OVERLAY_HEIGHT) - scrollWindow.getBoundingClientRect().top) : 'auto';
-      const topSpace = scrollWindow ? container.getBoundingClientRect().top - scrollWindow.getBoundingClientRect().top : 0;
-      this.setState({
-        topSpace,
-        bottomSpace
+export const OverlaySigil = (props: OverlaySigilProps): React.FC => {
+  const {
+    api,
+    className,
+    color,
+    contact,
+    group,
+    history,
+    onDismiss,
+    scrollWindow,
+    ship,
+    ...rest
+  } = { ...props };
+  const containerRef = useRef(null);
+  const [visible, setVisible] = useState<OverlaySigilState.visible>();
+  const [space, setSpace] = useState<OverlaySigilState.space>({
+    top: 'auto',
+    bottom: 'auto'
+  });
+
+  const updateContainerOffset = () => {
+    if (scrollWindow && containerRef && containerRef.current) {
+      const container = containerRef.current;
+      const bottomSpace = scrollWindow
+        ? scrollWindow.clientHeight -
+          (container.getBoundingClientRect().top +
+            OVERLAY_HEIGHT -
+            scrollWindow.getBoundingClientRect().top)
+        : 'auto';
+      const topSpace = scrollWindow
+        ? container.getBoundingClientRect().top -
+          scrollWindow.getBoundingClientRect().top
+        : 0;
+      setSpace({
+        ...space,
+        top: topSpace,
+        bottom: bottomSpace
       });
     }
-  }
-
-  componentWillUnmount() {
-    this.props.scrollWindow?.removeEventListener('scroll', this.updateContainerOffset, true);
-  }
-
-  render() {
-    const {
-      className,
-      ship,
-      contact,
-      color,
-      group,
-      history,
-      api,
-      sigilClass,
-      hideAvatars,
-      pr = 0,
-      pl = 0,
-      ...rest
-    } = this.props;
-
-    const { state } = this;
-
-    const img = (contact && (contact.avatar !== null) && !hideAvatars)
-      ? <BaseImage display='inline-block' src={contact.avatar} height={16} width={16} />
-      : <Sigil
-        ship={ship}
-        size={16}
+    setVisible(true);
+  };
+
+  useEffect(() => {
+    updateContainerOffset();
+    if (scrollWindow)
+      scrollWindow.addEventListener('scroll', updateContainerOffset);
+    return () => {
+      if (scrollWindow)
+        scrollWindow.removeEventListener('scroll', updateContainerOffset, true);
+    };
+  }, []);
```

Comment:
kinda wondering if this should have `scrollWindow` as a dependency, given that it looks like `this.window` can change in virtualscroller. cc: @tylershuster

Reply:
Sure, that makes sense. I made it optional because we also use this in notebooks and posts, where I don't think we virtualize the content and thus don't drill the scrollWindow ref all the way down. Totally open to revising.

Reply:
Yep, it should depend on that. There's already some weird behaviors (some resolved, others outstanding) when switching between chats due to this

Reply:
Just for clarity, should I leave as-is and make a separate task to pass the window ref down in other apps, or do that in this PR?

Reply:
The empty array `[]` at the end of the `useEffect` declaration should be `[scrollWindow]` instead — you're declaring that the effect declared depends on the identity of `scrollWindow` — if it changes, we want to recalculate the listeners because the element has changed.

Reply:
Understood, I realize now @liam-fitzgerald meant `this` as `this effect`, not `this component`. Thank you!